src/toil_lib/programs.py

```diff
@@ -15,34 +17,47 @@ def mock_mode():
     return True if int(os.environ.get('TOIL_SCRIPTS_MOCK_MODE', '0')) else False
 
 
-def docker_call(tool,
+def docker_call(job,
+                tool,
                 parameters=None,
                 work_dir='.',
                 rm=True,
+                detached=False,
                 env=None,
                 outfile=None,
                 inputs=None,
                 outputs=None,
                 docker_parameters=None,
                 check_output=False,
-                mock=None):
+                mock=None,
+                defer=None):
     """
     Calls Docker, passing along parameters and tool.
 
+    :param toil.Job.job job: The Job instance for the calling function.
     :param str tool: Name of the Docker image to be used (e.g. quay.io/ucsc_cgl/samtools)
     :param list[str] parameters: Command line arguments to be passed to the tool
     :param str work_dir: Directory to mount into the container via `-v`. Destination convention is /data
-    :param bool rm: Set to True to pass `--rm` flag.
+    :param bool rm: Should the container be run with the --rm flag (Should it be removed upon
+           container exit)? rm and detached are mutually exclusive in Docker.
+    :param bool detached: Should the container be run with the --detached flag (Should it be run in
+           detached mode)? rm and detached are mutually exclusive in Docker.
     :param dict[str,str] env: Environment variables to be added (e.g. dict(JAVA_OPTS='-Xmx15G'))
-    :param bool sudo: If True, prepends `sudo` to the docker call
     :param file outfile: Pipe output of Docker call to file handle
     :param list[str] inputs: A list of the input files.
     :param dict[str,str] outputs: A dictionary containing the outputs files as keys with either None
-                                  or a url. The value is only used if mock=True
+           or a url. The value is only used if mock=True
```

Comment:
That's not true, or? The existence of the outputs are validated if mock=False, right?